Source/WTF/wtf/CheckedPtr.h

```diff
@@ -184,10 +184,21 @@ class CheckedPtr {
         if (T* ptr = PtrTraits::unwrap(m_ptr); ptr) [[likely]]
             ptr->decrementCheckedPtrCount();
     }
-
     typename PtrTraits::StorageType m_ptr;
 };
 
+template<typename T, typename U = RawPtrTraits<T>, typename X, typename Y>
+inline CheckedPtr<T, U> static_pointer_cast(const CheckedPtr<X, Y>& p)
+{
+    return CheckedPtr<T, U>(static_cast<T*>(p.get()));
```

Comment:
I think we wanna do this static cast directly in uncheckedDowncast, etc... and suppress the unsafe cast warning on that line with `SUPPRESS_MEMORY_UNSAFE_CAST`